### PR TITLE
Convert service info functions to pybind11

### PR DESCRIPTION
**Pull request by cottsay**
Part of #665

Converts:
* rclpy_service_info_get_sequence_number
* rclpy_service_info_get_source_timestamp
* rclpy_service_info_get_received_timestamp

### CODE REVIEW

**Comment by cottsay** [comment]
CI (build: --packages-up-to rclpy test: --packages-select rclpy)
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/13796/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/8666/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/11507/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/13877/)

File: rclpy/src/rclpy/service_info.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Include pybind11 before rclpy_common/handle.h includes Python.h
+#include <pybind11/pybind11.h>
+
+#include <rmw/types.h>
+
+#include "rclpy_common/handle.h"
```
Comment (cottsay) [inline rclpy/src/rclpy/service_info.cpp:20]: I think I can drop this.

File: rclpy/src/rclpy/service_info.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+#include "service_info.hpp"
+
+namespace rclpy
+{
+int64_t
+service_info_get_sequence_number(py::capsule pyservice_info)
+{
+  auto service_info = static_cast<rmw_service_info_t *>(
+    PyCapsule_GetPointer(pyservice_info.ptr(), "rmw_service_info_t"));
```
Comment (sloretz) [inline rclpy/src/rclpy/service_info.cpp:30]: Can use the pybind11 api to get the pointer which will throw if getting the pointer fails.

```C++
auto service_info = static_cast<rmw_service_info_t *>(pyservice_info);
```

Downside is losing the name checking, but that can be done manually.

```C++
if (0 != strcmp("rmw_service_info_t" != pyservice_info.name()) {
  throw py::value_error("capsule is not an rmw_service_info_t");
}
auto service_info = static_cast<rmw_service_info_t *>(pyservice_info);
```

File: rclpy/src/rclpy/service_info.cpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  if (!service_info) {
```
Comment (sloretz) [inline rclpy/src/rclpy/service_info.cpp:31]: no need for this check if using the pybind11 API: https://github.com/pybind/pybind11/blob/3b1dbebabc801c9cf6f0953a4c20b904d444f879/include/pybind11/pytypes.h#L1191

File: rclpy/src/rclpy/service_info.cpp (hunk 1 above)
Comment (sloretz) [reply rclpy/src/rclpy/service_info.cpp:20]: definitely

File: rclpy/src/rclpy/service_info.cpp (hunk 3 above)
Comment (cottsay) [reply rclpy/src/rclpy/service_info.cpp]: Dropped in 7fa94c4.

File: rclpy/src/rclpy/service_info.cpp (hunk 1 above)
Comment (cottsay) [reply rclpy/src/rclpy/service_info.cpp]: Dropped in 7fa94c4.

File: rclpy/src/rclpy/service_info.cpp (hunk 2 above)
Comment (cottsay) [reply rclpy/src/rclpy/service_info.cpp]: Awesome! Applied in 7fa94c4.

**Review by sloretz — APPROVED** [review]
LGTM with green CI!

**Pull request merged**